File: sentinel-pay-be/payment-service/src/main/java/com/paymentservice/dto/PaymentResponse.java
package com.paymentservice.dto;

import com.paymentservice.enums.PaymentStatus;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Setter
@Builder
public class PaymentResponse {

    private Long paymentId;

    private Long senderUserId;

    private Long receiverUserId;

    private BigDecimal amount;

    private PaymentStatus status;

    private LocalDateTime createdAt;

}

File: sentinel-pay-fe/src/app/features/payment/model/payment-response.ts
import { PaymentStatus } from './payment-status';

export interface PaymentResponse {
  paymentId: number;
  senderUserId: number;
  receiverUserId: number;
  amount: number;
  status: PaymentStatus;
  createdAt: string;
  failureReason: string | null;
}
